File: leetcode/roman-to-integer.cpp
#include<bits/stdc++.h>

using namespace std;

class Solution {
public:
    int romanToInt(string s) {
        vector<int> t;
        for(char c: s){
            if(c=='I')t.push_back(1);
            else if(c=='V')t.push_back(5);
            else if(c=='X')t.push_back(10);
            else if(c=='L')t.push_back(50);
            else if(c=='C')t.push_back(100);
            else if(c=='D')t.push_back(500);
            else{t.push_back(1000);}
        }
        int sum = t[0];
        for(int x=1;x<t.size();x++){
            if(t[x-1]<t[x]){
                sum+=t[x]-2*(t[x-1]);
            }
            else{sum+=t[x];}
        }
        return sum;
    }
};

File: leetcode/roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        total = 0
        tr = []
        for x in s:
            if x == "I":
                tr.append(1)
            elif x == "V":
                tr.append(5)
            elif x == "X":
                tr.append(10)
            elif x == "L":
                tr.append(50)
            elif x == "C":
                tr.append(100)
            elif x == "D":
                tr.append(500)
            else:
                tr.append(1000)
        iterations = 0
        for x in range(0, len(tr)):
            m = 0
            if iterations == 0:
                m = tr[0]
                iterations += 1
            elif tr[x-1] < tr[x]:
                m = tr[x]-tr[x-1]-tr[x-1]
            else:
                m = tr[x]
            total += m


        return total